physical/raft/transaction.go

```diff
@@ -172,6 +197,15 @@ func (b *RaftBackend) newTransaction(ctx context.Context, writable bool) (*RaftT
 		return nil, fmt.Errorf("failed to start underlying bbolt transaction: %w", err)
 	}
 
+	index, beginValue, err := b.createBeginTxOpValue()
```

Comment:
Strictly I think we want this to be ordered above the begin call: we could stall between begin and fetching this index, causing us to not see fs entries (in the tx) not included by this index, meaning we could fast apply when we shouldn't commit at all. Conversely, the other ordering is safer as we could fail to fast apply it our index is too old but the tx matches, so we could fail to fast apply but verify correctly.